tests/smoke.sh

```diff
@@ -37,47 +43,65 @@ function start_and_wait_for_llama_stack_container {
 }
 
 function test_model_list {
-  for model in "$INFERENCE_MODEL" "$EMBEDDING_MODEL"; do
-    echo "===> Looking for model $model..."
-    resp=$(curl -fsS http://127.0.0.1:8321/v1/models)
+  # Check if model is provided
+  if [ -z "$1" ]; then
```

Comment:
Could you use the new `validate_model_parameter` method here to reduce duplication?